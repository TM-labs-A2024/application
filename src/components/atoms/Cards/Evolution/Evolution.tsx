import { Link } from '@chakra-ui/next-js'
import { Text } from '@chakra-ui/react'
import { Evolution as EvolutionType } from '@src/types'
import NextLink from 'next/link'
import React from 'react'

import Chevron from '../../../../../public/static/icons/chevron.svg'

export default function Evolution(evolution: EvolutionType) {
  const { href, title, description, comment } = evolution

  return (
    <Link as={NextLink} href={href} padding="0.5rem" className="w-full">
      <div className="mb-1 flex flex-row items-center justify-between">
        <div className="flex flex-col items-start gap-1">
          <Text fontSize="md" as="b">
            {title}
          </Text>
          <div className="flex flex-col items-start gap-1">
            <Text>{description}</Text>
            <Text>{comment}</Text>
          </div>
        </div>
        <Chevron />
      </div>
    </Link>
  )
}
