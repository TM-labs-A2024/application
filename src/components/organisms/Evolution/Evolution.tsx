import { ArrowBackIcon } from '@chakra-ui/icons'
import {
  IconButton,
  Text,
  Accordion,
  AccordionItem,
  AccordionButton,
  AccordionPanel,
  AccordionIcon,
  Box
} from '@chakra-ui/react'
import Profile from '@src/components/atoms/Cards/Profile'
import { Patient } from '@src/types'
import { isIOS } from '@utils/index'
import { useRouter } from 'next/navigation'
import React from 'react'

export default function Evolution({
  patient,
  goBackRef,
  title,
  data
}: {
  patient?: Patient
  goBackRef: string
  title: string
  data: {
    title: string
    content: string
  }[]
}) {
  // --- Hooks -----------------------------------------------------------------
  const router = useRouter()
  // --- END: Hooks ------------------------------------------------------------

  return (
    <div
      className={`flex h-screen w-screen flex-col p-8 lg:px-96 ${isIOS() ? 'pt-20' : 'pt-8'}`}
      data-testid="evolution"
    >
      <div className="mb-8 flex flex-row items-center justify-start">
        <IconButton
          size="xl"
          className="mr-4"
          aria-label="back"
          variant="link"
          icon={<ArrowBackIcon />}
          onClick={() => {
            router.push(goBackRef)
          }}
        />
        <Text className="font-medium">{title}</Text>
      </div>
      {patient && <Profile patient={patient} />}
      <div className="flex h-full flex-col overflow-scroll">
        <Accordion defaultIndex={[]} allowMultiple>
          {data?.map((item, idx) => (
            <AccordionItem key={`evolution-item-${idx + 1}`}>
              <h2>
                <AccordionButton>
                  <Box as="span" flex="1" textAlign="left">
                    {item.title}
                  </Box>
                  <AccordionIcon />
                </AccordionButton>
              </h2>
              <AccordionPanel pb={4}>{item.content}</AccordionPanel>
            </AccordionItem>
          ))}
        </Accordion>
      </div>
    </div>
  )
}
