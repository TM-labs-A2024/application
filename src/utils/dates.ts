export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
