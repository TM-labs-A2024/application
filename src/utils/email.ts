import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
