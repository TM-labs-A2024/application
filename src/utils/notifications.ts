import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
