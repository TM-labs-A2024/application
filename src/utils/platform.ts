import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'

export const isMobile = (window: { screen: { availWidth: number } }) =>
  Capacitor.getPlatform() === 'ios' ||
  Capacitor.getPlatform() === 'android' ||
  window?.screen?.availWidth < 425
