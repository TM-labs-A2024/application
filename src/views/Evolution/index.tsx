import Evolution from '@components/organisms/Evolution'
import { Patient } from '@src/types'
import React from 'react'

export default function EvolutionView({
  patient,
  goBackRef,
  title,
  data
}: {
  patient?: Patient
  goBackRef: string
  title: string
  data: {
    title: string
    content: string
  }[]
}) {
  return <Evolution patient={patient} goBackRef={goBackRef} title={title} data={data} />
}
