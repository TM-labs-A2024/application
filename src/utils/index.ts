import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
