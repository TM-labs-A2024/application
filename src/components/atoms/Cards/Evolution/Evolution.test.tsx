import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import Evolution from './Evolution'

const evolution = {
  href: '/evolucion/3fa85f64-5717-4562-b3fc-2c963f66afa5/1/1',
  title: 'Evolución 1: 17, septiembre 2024',
  description: 'Editado por: Pedro Pérez',
  comment: 'Patología: Dolor'
}

describe('Atoms > Cards > Evolution test', () => {
  test('The card shows the title, description and comments of the evolution', async () => {
    render(<Evolution {...evolution} />)

    expect(await screen.findByText(evolution.title)).toBeInTheDocument()
    expect(await screen.findByText(evolution.description)).toBeInTheDocument()
    expect(await screen.findByText(evolution.comment)).toBeInTheDocument()
  })

  test('The url matches the expected', async () => {
    render(<Evolution {...evolution} />)

    const enlace = await screen.findByRole('link')
    expect(enlace).toHaveAttribute('href', evolution.href)
  })

  test('Matches the snapshot', async () => {
    render(<Evolution {...evolution} />)

    expect(await screen.findByTestId('evolution-card')).toMatchSnapshot()
  })
})
