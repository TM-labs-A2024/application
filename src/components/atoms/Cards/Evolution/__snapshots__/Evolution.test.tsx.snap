// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > Cards > Evolution test Matches the snapshot 1`] = `
<a
  class="chakra-link w-full css-ofs0kf"
  data-testid="evolution-card"
  href="/evolucion/3fa85f64-5717-4562-b3fc-2c963f66afa5/1/1"
>
  <div
    class="mb-1 flex flex-row items-center justify-between"
  >
    <div
      class="flex flex-col items-start gap-1"
    >
      <b
        class="chakra-text css-qqfgvy"
      >
        Evolución 1: 17, septiembre 2024
      </b>
      <div
        class="flex flex-col items-start gap-1"
      >
        <p
          class="chakra-text css-0"
        >
          Editado por: Pedro Pérez
        </p>
        <p
          class="chakra-text css-0"
        >
          Patología: Dolor
        </p>
      </div>
    </div>
    <div />
  </div>
</a>
`
